### PR TITLE
Document `make backport` details and usage.

### PR DESCRIPTION
**Pull request by Jaanki**
Depends on https://github.com/submariner-io/shipyard/pull/618

Signed-Off-By: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
We don't have gitlint on this repo unfortunately, but if we did I think it would fail on the period on the first line of the commit message. Might still be best to follow those practices.

File: src/content/development/backports/_index.en.md [hunk 1]
```diff
@@ -26,37 +26,21 @@ This ensures that commit identifiers will remain stable during the backport proc
...
-Push the resulting branch and open a new pull request; add “Backport: ” to the pull request description to identify it as a backport.
-(Do not use the “backport” label here; that doesn’t identify backports, it identifies requested backports.)
-On the original pull request, add a comment referencing the new pull request and identifying it as a backport to the relevant release.
-
-Once _all_ the requested backports of the original pull request have been handled, add the “backport-handled” label.
-This will result in the pull request disappearing from the list of pending backports.
+Backporting a commit is automated by running `make backport release=<release-branch> pr=<PR to cherry-pick>`. The `make` target runs a script,
+[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh). The script is originally being used by Kubernetes community
+and has been reused, and edited to suit the needs, from [their repo](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
+The script does the following:
```
Comment (dfarrell07) [inline src/content/development/backports/_index.en.md]:
```suggestion
Backporting a commit is automated by running:

`make backport release=<release-branch> pr=<PR number to backport>`

The make target runs [`backport.sh`](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh, originally developed by the Kubernetes community as [`cherry_pick_pull.sh`](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).

The script does the following:
```

File: src/content/development/backports/_index.en.md [hunk 2]
```diff
@@ -26,37 +26,21 @@ This ensures that commit identifiers will remain stable during the backport proc
...
+1. Creates a PR on `<release-branch>` with title `Automated cherry pick of <original PR number> <original PR title>`.
+2. Adds `backport-handled` label to the original PR.
+3. Gives an option to create patches to a release branch without making a PR by setting `DRY_RUN` variable.
+   When `DRY_RUN` is set the script will leave you in a branch containing the commits you cherry-picked.
+4. Multiple PRs can be cherry-picked together by passing the comma-separated list of PRs to be cherry-picked like `pr=<PR1,PR2>`.   
+
+The script assumes following values. Please change them according to your setup.
+1. Remote for the upstream repository, `UPSTREAM_REMOTE`, is set to `origin`.
+2. Remote for your forked repository, `FORK_REMOTE`  is set to `GITHUB_USER`.
+3. `GITHUB_USER` needs to be set to your `GitHub username`.
```
Comment (dfarrell07) [inline src/content/development/backports/_index.en.md]:
```suggestion
3. `GITHUB_USER` needs to be set to your GitHub username.
```

**Review by dfarrell07 — CHANGES_REQUESTED** [review]

File: src/content/development/backports/_index.en.md [hunk 3]
```diff
@@ -26,37 +26,21 @@ This ensures that commit identifiers will remain stable during the backport proc
...
+Backporting a commit is automated by running `make backport release=<release-branch> pr=<PR to cherry-pick>`. The `make` target runs a script,
+[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh). The script is originally being used by Kubernetes community
+and has been reused, and edited to suit the needs, from [their repo](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
+The script does the following:
+
+1. Creates a PR on `<release-branch>` with title `Automated cherry pick of <original PR number> <original PR title>`.
+2. Adds `backport-handled` label to the original PR.
+3. Gives an option to create patches to a release branch without making a PR by setting `DRY_RUN` variable.
+   When `DRY_RUN` is set the script will leave you in a branch containing the commits you cherry-picked.
+4. Multiple PRs can be cherry-picked together by passing the comma-separated list of PRs to be cherry-picked like `pr=<PR1,PR2>`.
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
1. Cherry-picks the commits from the PR onto <remote branch>.
2. Creates a PR on `<release-branch>` with the title `Automated backport of <original PR number>: <original PR title>`.
3. Adds the `backport-handled` label to the original PR.

The DRY_RUN environment variable can be set to skip creating the PR. When set, it leaves you in a branch containing the commits that were cherry-picked.

Multiple PRs can be backported together by passing a comma-separated list of PR numbers, eg `pr=630,631`.   
```

File: src/content/development/backports/_index.en.md [hunk 4]
```diff
@@ -26,37 +26,21 @@ This ensures that commit identifiers will remain stable during the backport proc
...
+
+1. Creates a PR on `<release-branch>` with title `Automated cherry pick of <original PR number> <original PR title>`.
+2. Adds `backport-handled` label to the original PR.
+3. Gives an option to create patches to a release branch without making a PR by setting `DRY_RUN` variable.
+   When `DRY_RUN` is set the script will leave you in a branch containing the commits you cherry-picked.
+4. Multiple PRs can be cherry-picked together by passing the comma-separated list of PRs to be cherry-picked like `pr=<PR1,PR2>`.   
+
+The script assumes following values. Please change them according to your setup.
+1. Remote for the upstream repository, `UPSTREAM_REMOTE`, is set to `origin`.
+2. Remote for your forked repository, `FORK_REMOTE`  is set to `GITHUB_USER`.
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
The script uses the following environment variables. Please change them according to your setup.
* `UPSTREAM_REMOTE`: the remote for the upstream repository. Defaults to `origin`.
* `FORK_REMOTE`: the remote for your forked repository. Defaults to `GITHUB_USER`.
* `GITHUB_USER`: needs to be set to your GitHub username.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/development/backports/_index.en.md [hunk 5]
```diff
@@ -26,37 +26,28 @@ This ensures that commit identifiers will remain stable during the backport proc
 
 ### Standalone pull requests
 
-Update your local clone of the relevant repository, and identify the commit(s) which need to be backported
-(all the commits listed in the pull request, as merged).
-If multiple commits are involved, it is expected that they will be consecutive (since they were part of the same pull request).
+Backporting a commit is automated by running:
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
Backporting a pull request (PR) is automated by running:
```

File: src/content/development/backports/_index.en.md [hunk 6]
```diff
@@ -26,37 +26,28 @@ This ensures that commit identifiers will remain stable during the backport proc
...
-Check out a new backport branch based on the _target_ branch;
-for example, to backport PR 1431 in the operator to the 0.9 release branch:
+`make backport release=<release-branch> pr=<PR to cherry-pick>`
 
-```sh
-git checkout -b backport-1431-0.9 origin/release-0.9
-```
+The `make` target runs a script,
+[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh), originally being used by Kubernetes community
+and has been reused, and edited to suit the needs, from [their repo](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh), originally developed by the [Kubernetes community](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
```

File: src/content/development/backports/_index.en.md [hunk 7]
```diff
@@ -26,37 +26,28 @@ This ensures that commit identifiers will remain stable during the backport proc
...
 
-Then, cherry pick the commits:
+The script does the following:
 
-```sh
-git cherry-pick -x 0448fcd60972da03c84c345c24c2cf1c34b6b42b
-```
+1. Cherry-picks the commits from the PR onto <remote branch>.
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
1. Cherry-picks the commits from the PR onto `<remote branch>`.
```

File: src/content/development/backports/_index.en.md [hunk 8]
```diff
@@ -26,37 +26,28 @@ This ensures that commit identifiers will remain stable during the backport proc
...
+2. Creates a PR on `<release-branch>` with the title `Automated backport of <original PR number>: <original PR title>`.
+3. Adds the `backport-handled` label to the original PR and `automated-backport` label to the backported PR.
```
Comment (tpantelis) [inline src/content/development/backports/_index.en.md]:
```suggestion
3. Adds the `backport-handled` label to the original PR and the `automated-backport` label to the backported PR.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]
Just a few more minor things.

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**